### ISSUE TITLE
[Feature]: Urls/Links Receiver

### ISSUE THREAD

**Issue by ELWAER-M** [termux/termux-app#3024]
### Feature description

termux already have a share/view file receiver, a url receiver will be a nice feature to have too, we will be able to use termux to download files using `wget` or `aria2`, or checking the links before sending them to the browser, that will open a lot of possibilities!

### Additional information

none

**Comment by 2096779623** [comment]
`termux-download --help`

**Comment by ELWAER-M** [comment]
> Isn't this feature same as described in [wiki.termux.com/wiki/Intents_and_Hooks](https://wiki.termux.com/wiki/Intents_and_Hooks) (termux-url-opener hook)?

i mean like when you open a youtube link, android will ask you if you want to open it in the youtube app or in a browser, without need to share the link

**Comment by tareksander** [comment]
I don't think this feature is possible in general. It requires the use of [Android app links](https://developer.android.com/training/app-links#android-app-links) for which you need to declare each domain name separately and the domain needs to host a file that makes your app authorized to be a handler for these links. Since Android 12 you can manually assign apps to be the default handler without the website approval, but that is likely off-limits to Termux, since it's stuck targeting Android 9.

**Comment by agnostic-apollo** [comment]
> I don't think this feature is possible in general.

App links would be needed for say if we wanted termux.dev to always open with termux app, or how facebook, twitter links would open with their apps. Random web links can be received by any app when clicked, like how browsers apps receive and open links. Only a receiver with `intent-filter` for `http` and `https` schemes would be required, in which a script is executed, like `termux-file-editor` is. It can be disabled by default as well and enabled if users want it, as per af6ac30bb. 

https://github.com/chromium/chromium/blob/6bb4aac8006c845f61d711d9adc08165daca24ae/chrome/android/java/AndroidManifest.xml#L224

**Comment by azmras** [comment]
I just made a script for me..

$cat /data/data/com.termux/files/usr/bin/ytv

#!/data/data/com.termux/files/usr/bin/sh
if [ $# != 1 ]; then
	echo 'usage: ytv <video-id>'
	echo 'Open video in youtube app.'
	exit 1
fi
**am start --user 0 -n com.google.android.youtube.tv/com.google.android.apps.youtube.tv.activity.ShellActivity -d "https://youtu.be/${1}"**
